web/index.js

```diff
@@ -1,403 +1,174 @@
-var hasher = crypto.createHash('sha256');
-hasher.update(os.hostname());
-var hostUid = hasher.digest('hex') + '-' + uid;
-server.method('getUid', function() { return hostUid });
-
-function checkHapiPluginError (name) {
-  return function (error) {
-    if (error) {
-      console.error('Failed loading a Hapi plugin: "' + name + '".');
-      throw error;
-    }
-  };
-}
-
-// Set up HAPI
-server.connection({
-  port: port,
-  // According to the HTTP spec and Chrome audit tool, Cache-Control headers should match what
-  // would be sent for 200 when a 304 (Not Modified) is sent.
-  routes: {
-    cache: { statuses: [200,304] },
-    cors: { origin: [ hostWithPort, 'https://changex.org', 'https://coderdojo.com', 'http://localhost'], credentials: true }
-  }
-});
-
-if ('production' === env || 'staging' === env) {
-  server.ext('onRequest', function(request, reply) {
-    if (request.headers['x-forwarded-proto'] != 'https') {
-      return reply.redirect('https://' + request.headers.host + request.path);
-    }
-    reply.continue();
+exports.start = function () {
+  const availableLocales = new locale.Locales(_.pluck(languages, 'code'));
+  const server = new hapi.Server(options.hapi);
+  const port = process.env.PORT || 8000;
+  const host = process.env.HOSTNAME || '127.0.0.1';
+  const protocol = process.env.PROTOCOL || 'http';
+  const hostWithPort = `${protocol}://${host}:${port}`;
+  const uid = cuid();
+  const hasher = crypto.createHash('sha256');
+  hasher.update(os.hostname());
+  const hostUid = `${hasher.digest('hex')}-${uid}`;
+  server.method('getUid', () => hostUid);
+
+  // Set up HAPI
+  server.connection({
+    port,
+    // According to the HTTP spec and Chrome audit tool, Cache-Control headers should match what
+    // would be sent for 200 when a 304 (Not Modified) is sent.
+    routes: {
+      cache: { statuses: [200, 304] },
+      cors: { origin: [hostWithPort, 'https://changex.org', 'https://coderdojo.com', 'http://localhost'], credentials: true },
+    },
   });
-}
-
-server.register(inert, function (err) {
-  checkHapiPluginError('inert')(err);
-});
 
-server.register(require('blipp'), function (err) {
-  checkHapiPluginError('blipp')(err);
-});
+  server.register(
+    [
+      { register: vision },
+      { register: inert },
+      { register: blipp },
+      { register: ip2country,
+        routes: {
+          prefix: '/api/2.0/ip-country-details',
+        },
+      },
+      { register: cdAuth },
+      { register: hapiEtags, options: { varieties: ['plain', 'buffer', 'stream'] } },
+      { register: scooterAndBlankie },
+      { register: logging },
+      { register: swagger },
+      { register: chairo, options },
+      { register: apis },
+      { register: senecaPreloaders,
+        options: { handlers: ['seneca-event-preloader', 'seneca-dojo-preloader'] } },
+      { register: controllers },
+      { register: cpZenFrontend },
+    ],
+  )
+    .then(() => {
+      server.start((err) => {
```

Comment:
why mix callback and promises when server.start returns a promise?

Reply:
lazyness spotted, will fix